components/PrivatePlaylistMenu.vue

```diff
@@ -0,0 +1,184 @@
+<script lang="ts" setup>
+import type { NuxtIconName } from "#build//nuxt-icons";
+import type { RoutesNamedLocations } from "#build/typed-router";
+import { TrackCollectionApi } from "@bcc-code/bmm-sdk-fetch";
+import type {
+  GetTrackCollectionModel,
+  TrackModel,
+  TrackReference,
+} from "@bcc-code/bmm-sdk-fetch";
+import { Menu, MenuButton, MenuItem, MenuItems } from "@headlessui/vue";
+
+const { t } = useI18n();
+
+const props = withDefaults(
+  defineProps<{
+    playlist: GetTrackCollectionModel;
+    buttonClass?: string;
+  }>(),
+  {
+    buttonClass: "",
+  },
+);
+const emit = defineEmits<{ "playlist-changed": [] }>();
+
+type DropdownMenuItem = {
+  text: string;
+  icon?: NuxtIconName;
+} & ({ link: RoutesNamedLocations } | { clickFunction: Function });
+
+const showEditDialog = ref(false);
+const showDeleteDialog = ref(false);
+const playlistName = ref("");
+
+const tracksToTrackReferences = (tracks: TrackModel[] | undefined | null) => {
+  const list: TrackReference[] = [];
+  if (!tracks) return list;
+
+  tracks.forEach((track) => {
+    if (track.id && track.language)
+      list.push({
+        id: track.id,
+        language: track.language,
+      });
+  });
+  return list;
+};
+
+const savePlaylist = async () => {
+  showEditDialog.value = false;
+  await new TrackCollectionApi().trackCollectionIdPut({
+    id: props.playlist.id,
+    updateTrackCollectionCommand: {
+      id: props.playlist.id,
+      name: playlistName.value,
+      trackReferences: tracksToTrackReferences(props.playlist.tracks),
+    },
+  });
+  emit("playlist-changed");
+  refreshPrivatePlaylists();
+};
+const deletePlaylist = async () => {
+  navigateTo();
+  await new TrackCollectionApi().trackCollectionIdDelete({
+    id: props.playlist.id,
+  });
+  refreshPrivatePlaylists();
+};
+
+const dropdownMenuItems = () => {
+  const items: DropdownMenuItem[] = [];
+
+  items.push({
+    // icon: "icon.ai", // ToDo: get comment icon
```

Comment:
@SimonSimCity @sifferhans I think Søren added that icon fairly recently. How do I get new icons?

Reply:
You only have to add a file to the `assets/icons` folder. The nuxt-plugin should pick up that you added a file and update the typescript definition.

Reply:
done